patra_toolkit/patra_model_card.py

```diff
@@ -332,21 +335,19 @@ def _get_hash_id(self, patra_server_url):
             patra_server_url (str): The Patra server URL used to generate the ID.
 
         Returns:
-            str: A unique hash identifier for the model card.
+            str: A unique identifier for the model card.
         """
-        combined_string = f"{self.name}:{self.version}:{self.author}"
         try:
             if patra_server_url:
                 patra_hash_url = f"{patra_server_url}/get_hash_id"
                 headers = {'Content-Type': 'application/json'}
-                response = requests.get(patra_hash_url, params={"combined_string": combined_string}, headers=headers)
+                response = requests.get(patra_hash_url,
+                                        params={"author": self.author, "name": self.name, "version": self.version},
+                                        headers=headers)
                 response.raise_for_status()
                 return response.json()
-            else:
-                return hashlib.sha256(combined_string.encode()).hexdigest()
         except requests.exceptions.RequestException as e:
-            print("Could not connect to the Patra Server, generating the ID locally")
-            return hashlib.sha256(combined_string.encode()).hexdigest()
+            return {"error": f"An unexpected error occurred: {str(e)}"}
```

Comment:
You are returning a string from the try catch in case of an exception. You need to raise an error so the function calling this function can handle it. And make the error meaningful and use more custom messages to specify which exactly the error is.